hydromt_sfincs/utils.py

```diff
@@ -958,3 +959,55 @@ def _dist(a, b):
         nmax = int(np.ceil(axis2 / res))
 
     return x0, y0, mmax, nmax, rot
+
+
+def build_overviews(
+    fn: Union[str, Path],
+    resample_method: str = "average",
+    overviews: Union[list, str] = "auto",
+    logger=logger,
+):
+    """Build overviews for GeoTIFF file.
+
+    Overviews are reduced resolution versions of your dataset that can speed up
+    rendering when you don’t need full resolution. By precomputing the upsampled
+    pixels, rendering can be significantly faster when zoomed out.
+
+    Parameters
+    ----------
+    fn : str, Path
+        Path to GeoTIFF file.
+    method: str
+        Resampling method, by default "average". Other option is "nearest".
+    overviews: list of int, optional
+        List of overview levels, by default "auto". When set to "auto" the
+        overview levels are determined based on the size of the dataset.
+    """
+
+    # check if fn is a geotiff file
+    extensions = [".tif", ".tiff"]
+    assert any(
+        fn.endswith(ext) for ext in extensions
+    ), f"File {fn} is not a GeoTIFF file."
+
+    # open rasterio dataset
+    with rasterio.open(fn, "r+") as src:
+        # determine overviews when not provided
+        if overviews == "auto":
+            max_level = get_maximum_overview_level(src.width, src.height, 256)
```

Comment:
I would suggest to not hardcode the 256 but try to read it from the profile blockxsize argument.

```suggestion
            bs = src.profile.get('blockxsize', 256)
            max_level = get_maximum_overview_level(src.width, src.height, bs)
```

Reply:
please check if this code works before accepting, didn't test it!

Reply:
Works in my example, thanks for the suggestion!